### PR TITLE
Autotest: Added functions to arducopter autotest for motor failure testing

### PR DESCRIPTION
**Pull request by GuilhermeGSousa**
This PR firstly extends the changes made to sim_vehicle.py made in #5914 to include different parameter files by default on the arducopter autotest function. It also includes a function, using the changes from this same PR, I am using to test motor failure on hexacopters and quadcopters.

### CODE REVIEW

File: Tools/autotest/arducopter.py [hunk 1]
```diff
@@ -924,7 +924,33 @@ def fly_mission(mavproxy, mav, height_accuracy=-1.0, target_altitude=None):
     wait_mode(mav, 'LOITER')
     return ret
 
-
+def fly_motor_fail(mavproxy, mav, fail_servo=0, fail_mul=1.0, holdtime=30):
+    """Fly a mission from a file."""
```
Comment (peterbarker) [inline Tools/autotest/arducopter.py]: Incorrect comment

File: Tools/autotest/arducopter.py [hunk 2]
```diff
@@ -924,7 +924,33 @@ def fly_mission(mavproxy, mav, height_accuracy=-1.0, target_altitude=None):
...
+    global homeloc
```
Comment (peterbarker) [inline Tools/autotest/arducopter.py]: Unneeded globals

File: Tools/autotest/arducopter.py [hunk 3]
```diff
@@ -924,7 +924,33 @@ def fly_mission(mavproxy, mav, height_accuracy=-1.0, target_altitude=None):
...
+    global num_wp
+    if (fail_servo < 0) or (fail_servo > 6):
```
Comment (peterbarker) [inline Tools/autotest/arducopter.py]: Raise an exception on invalid input instead of silently fixing it.

File: Tools/autotest/arducopter.py [hunk 4]
```diff
@@ -924,7 +924,33 @@ def fly_mission(mavproxy, mav, height_accuracy=-1.0, target_altitude=None):
...
+        fail_servo = 0
+    mavproxy.send('switch 5\n')  # loiter mode
+    wait_mode(mav, 'LOITER')
+    print("test: Fly a mission from 1 to {0}, with motor {1} efficiency at {2}%".format(num_wp,fail_servo+1,fail_mul))
```
Comment (peterbarker) [inline Tools/autotest/arducopter.py]: Not flying a mission :-)

File: Tools/autotest/arducopter.py [hunk 5]
```diff
@@ -924,7 +924,33 @@ def fly_mission(mavproxy, mav, height_accuracy=-1.0, target_altitude=None):
...
+        data = mav.recv_match(type='SERVO_OUTPUT_RAW', blocking=True)
+        servo_list=[data.servo1_raw, data.servo2_raw, data.servo3_raw, data.servo4_raw, data.servo5_raw, data.servo6_raw, data.servo7_raw, data.servo8_raw]
+        output_message = "PWM output per motor: "
+        for i, val in enumerate(servo_list):
+            if val>0: #Prevent printing of unused servo channels
+                if i==fail_servo:
+                    output_message+="(failed) "
+                output_message+= "servo " + str(i+1) + " " + str(val) + "; "
+        print(output_message)
+    #Reset motor efficiency
```
Comment (peterbarker) [inline Tools/autotest/arducopter.py]: Should the motor be un-failed as well?

File: Tools/autotest/arducopter.py [hunk 6]
```diff
@@ -972,7 +998,30 @@ def fly_ArduCopter(binary, viewerip=None, use_map=False, valgrind=False, gdb=Fal
 
     # setup test parameters
     if params_file is None:
-        params_file = "{testdir}/default_params/copter.parm"
```
Comment (peterbarker) [inline Tools/autotest/arducopter.py]: This is a distinct change and deserves a separate commit, if not a separate PR.

**Comment by GuilhermeGSousa** [comment]
I completely forgot to change a few things here, I initially had it to follow a waypoint mission with a failed motor, I'll make the changes for this test. As for the parameter changes should I remove them from this PR and make a new one?

File: Tools/autotest/arducopter.py (hunk 5 above)
Comment (GuilhermeGSousa) [inline Tools/autotest/arducopter.py]: In this case SIM_ENGINE_FAIL only defines the servo to which the efficiency multiplier SIM_ENGINE_MUL is applied to, so un-failing the motor comes down to set the multiplier to one, but I can also reset SIM_ENGINE_FAIL to a given servo (0 for example)

Comment (GuilhermeGSousa) [inline Tools/autotest/arducopter.py]: Also I noticed that SIM_ENGINE_FAIL 0 corresponds to the servo 1, maybe I should change it so that SIM_ENGINE_FAIL corresponds to the servo
